src/python/pants/engine/legacy/address_mapper.py

```diff
@@ -75,7 +76,10 @@ def _internal_scan_specs(self, specs, fail_fast=True, missing_is_fatal=True):
 
     addresses = set()
     for (spec, _), state in root_entries.items():
-      if missing_is_fatal and not state.value.dependencies:
+      if isinstance(state, Throw) and missing_is_fatal:
```

Comment:
Hm. So, you're saying I should go with the original behavior unless I know the exception is one of the resolve related ones? If I did that, then other exceptions would not be surfaced to the user.

Maybe it'd be better to not have a type match, but include the type in the resulting message?